x-pack/plugins/security_solution/public/flyout/document_details/left/components/add_note.tsx

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo, useCallback, useEffect, useState } from 'react';
+import {
+  EuiButton,
+  EuiComment,
+  EuiCommentList,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSpacer,
+} from '@elastic/eui';
+import { useDispatch, useSelector } from 'react-redux';
+import { i18n } from '@kbn/i18n';
+import { ADD_NOTE_BUTTON_TEST_ID, ADD_NOTE_MARKDOWN_TEST_ID } from './test_ids';
+import { useAppToasts } from '../../../../common/hooks/use_app_toasts';
+import type { State } from '../../../../common/store';
+import {
+  createNoteByDocumentId,
+  ReqStatus,
+  selectCreateNoteByDocumentIdError,
+  selectCreateNoteByDocumentIdStatus,
+} from '../../../../notes/store/notes.slice';
+import { MarkdownEditor } from '../../../../common/components/markdown_editor';
+
+export const ADD_NOTE_BUTTON = i18n.translate('xpack.securitySolution.notes.addNoteButtonLabel', {
+  defaultMessage: 'Add note',
+});
+export const CREATE_NOTE_ERROR = i18n.translate(
+  'xpack.securitySolution.notes.createNoteErrorLabel',
+  {
+    defaultMessage: 'Error create note',
+  }
+);
+
+export interface AddNewNoteProps {
+  /**
+   * Id of the document
+   */
+  eventId: string;
+}
+
+/**
+ * Renders a markdown editor and a add button to create new notes
+ */
+export const AddNote = memo(({ eventId }: AddNewNoteProps) => {
+  const dispatch = useDispatch();
+  const { addError: addErrorToast } = useAppToasts();
+  const [editorValue, setEditorValue] = useState('');
+
+  const createStatus = useSelector((state: State) => selectCreateNoteByDocumentIdStatus(state));
+  const createError = useSelector((state: State) => selectCreateNoteByDocumentIdError(state));
+
+  const addNote = useCallback(() => {
+    dispatch(
+      createNoteByDocumentId({
+        note: {
+          timelineId: '',
+          eventId,
+          note: editorValue,
+        },
+      })
+    );
+    setEditorValue('');
+  }, [dispatch, editorValue, eventId]);
+
+  useEffect(() => {
+    if (createStatus === ReqStatus.Failed && createError) {
+      addErrorToast(null, {
+        title: CREATE_NOTE_ERROR,
+      });
+    }
+  }, [addErrorToast, createError, createStatus]);
+
+  return (
+    <>
+      <EuiCommentList>
+        <EuiComment username="">
+          <MarkdownEditor
+            dataTestSubj={ADD_NOTE_MARKDOWN_TEST_ID}
+            value={editorValue}
+            onChange={setEditorValue}
```

Comment:
is there an uncontrolled variant? typing something as it is wil re-render the whole `AddNote` component :D

Reply:
this is a mistake on my end, I meant to fix it before creating the PR but forgot. Thanks, fixed in this [last commit](https://github.com/elastic/kibana/pull/186807/commits/5c528c4996a900d0aabc3acdb1d18526bd0613ec)